plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/AbstractTrinoCatalog.java

```diff
@@ -254,6 +255,23 @@ protected Map<String, String> createViewProperties(ConnectorSession session)
                 .buildOrThrow();
     }
 
+    protected <T> SnapshotUpdate<T> setSnapshotSummaryExtraMetadata(ConnectorSession session, SnapshotUpdate<T> update)
+    {
+        extraSummaryMetadata(session)
+                .forEach(update::set);
+        // return the update for method chaining.
+        return update;
+    }
+
+    protected Map<String, String> extraSummaryMetadata(ConnectorSession session)
+    {
+        return ImmutableMap.<String, String>builder()
+                .put(TRINO_CREATED_BY, TRINO_CREATED_BY_VALUE)
+                .put(PRESTO_VERSION_NAME, trinoVersion)
+                .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
```

Comment:
For new metadata, why would we want to use Presto prefix?

Reply:
I used this copying from the view iirc. Will replace it!

Reply:
did you intent do apply some changes here?

Reply:
This isn't quite used yet but I'll update it now anyway.s

I am still using the method from within `IcebergMetadata`. I'd like to use this but I need to figure out a better way to organize it since it's needed in `IcebergMetadata` for now. Maybe a utility class?

I'll update the `PRESTO_*` out of it now though.